typescript/cosmos-sdk/src/tests/1_interchain_security.e2e-test.ts

```diff
@@ -0,0 +1,166 @@
+import { expect } from 'chai';
+import { step } from 'mocha-steps';
+
+import {
+  MerkleRootMultisigISM,
+  MessageIdMultisigISM,
+} from '../../../cosmos-types/dist/types/hyperlane/core/interchain_security/v1/types.js';
+import {
+  bytes32ToAddress,
+  isValidAddressEvm,
+} from '../../../utils/dist/addresses.js';
+import { SigningHyperlaneModuleClient } from '../index.js';
+
+import { createSigner } from './utils.js';
+
+describe('1. cosmos sdk interchain security e2e tests', async function () {
+  this.timeout(100_000);
+
+  let signer: SigningHyperlaneModuleClient;
+
+  before(async () => {
+    signer = await createSigner('alice');
+  });
+
+  step('create new NOOP ISM', async () => {
+    // ARRANGE
+    let isms = await signer.query.interchainSecurity.Isms({});
+    expect(isms.isms).to.be.empty;
+
+    // ACT
+    const txResponse = await signer.createNoopIsm({});
+
+    // ASSERT
+    expect(txResponse.code).to.equal(0);
+
+    const noopIsm = txResponse.response;
+
+    expect(noopIsm.id).to.be.not.empty;
+    expect(isValidAddressEvm(bytes32ToAddress(noopIsm.id))).to.be.true;
+
+    isms = await signer.query.interchainSecurity.Isms({});
+    expect(isms.isms).to.have.lengthOf(1);
+
+    let ism = await signer.query.interchainSecurity.Ism({
+      id: noopIsm.id,
+    });
+    expect(ism.ism?.type_url).to.equal(
+      '/hyperlane.core.interchain_security.v1.NoopISM',
+    );
+
+    let decodedIsm = await signer.query.interchainSecurity.DecodedIsm({
+      id: noopIsm.id,
+    });
+    expect(decodedIsm.ism.id).to.equal(noopIsm.id);
+    expect(decodedIsm.ism.owner).to.equal(signer.account.address);
+  });
+
+  step('create new MessageIdMultisig ISM', async () => {
+    // ARRANGE
+    let isms = await signer.query.interchainSecurity.Isms({});
+    expect(isms.isms).to.have.lengthOf(1);
+
+    const threshold = 2;
+    const validators = [
+      '0x3C24F29fa75869A1C9D19d9d6589Aae0B5227c3c',
+      '0xf719b4CC64d0E3a380e52c2720Abab13835F6d9c',
+      '0x98A56EdE1d6Dd386216DA8217D9ac1d2EE7c27c7',
+    ];
+
+    // note that the validators need to be sorted alphabetically
```

Comment:
What happens if they are not sorted?

Reply:
this is a requirement from the chain module side, if we would submit them not sorted the tx would fail. Testing such a thing would be more on the chain side instead of the TS sdk side